app/routes/_auth+/signup.tsx

```diff
@@ -128,12 +126,15 @@ export const meta: V2_MetaFunction = () => {
 }
 
 export default function SignupRoute() {
-	const data = useLoaderData<typeof loader>()
 	const signupFetcher = useFetcher<typeof action>()
-	const { form, fields } = useForm({
-		name: 'signup-form',
-		fieldMetadatas: data.fields,
-		errors: signupFetcher.data?.errors,
+	const [form, fields] = useForm({
+		id: 'signup-form',
+		constraint: getFieldsetConstraint(signupSchema),
+		lastSubmission: signupFetcher.data?.submission,
+		onValidate({ formData }) {
+			return parse(formData, { schema: signupSchema })
```

Comment:
Async validations in schemas are not supported in the client.
We can use a schema creator pattern as mentioned on the documentation [here](https://conform.guide/validation#async-validation).

Reply:
Done.